bigchaindb/backend/mongodb/query.py

```diff
@@ -17,7 +17,10 @@
 @register_query(MongoDBConnection)
 def write_transaction(conn, signed_transaction):
     try:
-        return conn.db['backlog'].insert_one(signed_transaction)
+        return conn.run(
+            lambda db:
```

Comment:
> not sure on how this can work.

Actually it doesnt even need lambda:

```python
@register_query(MongoDBConnection)
@retrylogic
def get_transaction_from_block(conn, transaction_id, block_id):
    return conn.query("get block please")
```

`@retrylogic` then handles stale connections.

OR

Take advantage of single dispatch to apply retry logic.